Main.java

```diff
@@ -3,58 +3,51 @@
+            System.out.println(Constants.errorMessage1);
+            DatabaseLink databaseLink = new DatabaseLink(textFilePath, searchKeyword, Constants.FAILURE ,Constants.errorMessage1, 0);
+            databaseLink.start();
             return;
         }
-        textFilePath = args[0];
-        searchKeyword = args[1];
-        File file = new File(textFilePath);
-        System.out.println("Processing...");
-        try {
-            if (isValidFile(file, textFilePath, searchKeyword)) {
+            textFilePath = args[0];
+            searchKeyword = args[1];
+            File file = new File(textFilePath);
+            System.out.println("Processing...");
+            if(isValidFile(file, textFilePath, searchKeyword)) {
                 if (file.length() != 0) {
                     wordCount(textFilePath, searchKeyword);
                 } else {
-                    errorMessage = "File does not contain any data";
-                    System.out.println(errorMessage);
-                    Database.databaseOperation(textFilePath, searchKeyword, searchResult = "Failure", errorMessage, 0);
+                    Constants.errorMessage2 = "File does not contain any data";
+                    System.out.println(Constants.errorMessage2);
+                    DatabaseLink databaseLink = new DatabaseLink(textFilePath, searchKeyword, Constants.FAILURE,Constants.errorMessage2, 0);
+                    databaseLink.start();
                 }
             }
-        } catch (ClassNotFoundException e) {
-            e.printStackTrace();
         }
-    }
 
-    private static boolean isValidFile(File file, String textFilePath, String searchKeyword) throws ClassNotFoundException {
-        String errorMessage;
+        private static boolean isValidFile(File file, String textFilePath, String searchKeyword) {
         if (!file.exists()) {
-            errorMessage = "File does not exist";
-            System.out.println(errorMessage);
-            Database.databaseOperation(textFilePath, searchKeyword, searchResult = "Failure", errorMessage, 0);
+            Constants.errorMessage3 = "File does not exist";
+            System.out.println(Constants.errorMessage3);
+            DatabaseLink databaseLink = new DatabaseLink(textFilePath, searchKeyword, Constants.FAILURE, Constants.errorMessage3, 0);
+            databaseLink.start();
             return false;
-        } else if (!(file.getName().endsWith(FILE_TYPE_TXT) || file.getName().endsWith(FILE_TYPE_JSON))) {
-            errorMessage = "File format not supported";
-            System.out.println(errorMessage);
-            Database.databaseOperation(textFilePath, searchKeyword, searchResult = "Failure", errorMessage, 0);
+        } else if (!(file.getName().endsWith(Constants.FILE_TYPE_TXT) || file.getName().endsWith(Constants.FILE_TYPE_JSON))) {
+            Constants.errorMessage4 = "File format not supported";

```

Comment:
You have already defined these messages as constants, then why are you reassigning them?

Reply:
Mistake from my part. Will remove them immediately.